services/provider/server/consumer.go

```diff
@@ -0,0 +1,157 @@
+package server
+
+import (
+	"context"
+	"fmt"
+
+	ocsv1alpha1 "github.com/red-hat-storage/ocs-operator/api/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type consumerCache struct {
```

Comment:
Naming is important as it can help avoid misunderstanding around usage and improve readability. , 
Why use the store if you do not hold any actual entities in the store?

Don't you think something like `consumerById` and `consumerByTicket` fields will be better?

Reply:
using `nameByUID` and `nameByTicket`.